validation_ingress.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"fmt"
+	extv1beta1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"regexp"
+	"strings"
+)
+
+var ingressHostRegExp = regexp.MustCompile(`^([a-zA-Z0-9_][a-zA-Z0-9_-]{0,62})(\.[a-zA-Z0-9_][a-zA-Z0-9_-]{0,62})*[._]?$`)
+var ingressPathRegExp = regexp.MustCompile(`^[A-zA-Z0-9_/.\-]*$`)
+
+type TlsDefinition struct {
+	host             string
+	secretName       string
+	ingressName      string
+	ingressNamespace string
+}
+
+type PathDefinition struct {
+	host             string
+	path             string
+	serviceName      string
+	servicePort      string
+	ingressName      string
+	ingressNamespace string
+}
+
+func (pathDefinition *PathDefinition) toUri() string {
+	return pathDefinition.host + pathDefinition.path
+}
+
+func ValidateIngress(validation *objectValidation, ingress *extv1beta1.Ingress, config *config) error {
+
+	targetDesc := fmt.Sprintf("Ingress %s.%s: ", ingress.Name, ingress.Namespace)
+
+	remoteIngresses, err := ingressClient().List(metav1.ListOptions{})
```

Comment:
Two questions about `.List()` here:
* does it list ingresses in all namespaces? (maybe we should have a test that spans multiple namespaces)
* won't this be really ineffective on a large cluster? How about using an `Informer` to watch for ingress adds/updates, instead of re-querying the whole thing every time? More on informers e.g. here: https://medium.com/firehydrant-io/stay-informed-with-kubernetes-informers-4fda2a21da9e I know that it's an optimization, but one that I believe we'll have to do soon (not necessarily in this version though).

Reply:
I made it more obvious that all namespaces are considered